Nodeletise laser_filters

Patrick-Lascombe opened the pull request:
By porting to nodelets scan_to_scan_filter_chain and scan_to_cloud_filter_chain we save the computation time of serializing and deserializing of data.

Patrick-Lascombe committed:
Nodolet transformation

Patrick-Lascombe committed:
Adding example launch and yaml for nodelet

doisyg committed:
fix nh topic visibility

Patrick-Lascombe committed:
ScanToCloudFilterChain is working as a nodelet

Patrick-Lascombe committed:
Adding dynamic reconfigure to angular_bound_filter

Patrick-Lascombe committed:
Correct dynamic reconfigure and add it to angular_bounds_filter_in_place

Patrick-Lascombe committed:
Dynamic reconfigure : new default and min/max parameter

Patrick-Lascombe committed:
Allow that min angle > max angle

If we want to crop points that are around the 0 rad of the LIDAR with only one filter, this update makes it possible

Patrick-Lascombe committed:
#132 Close

Patrick-Lascombe committed:
Adding dynamic reconfigure to intensity_filter

Patrick-Lascombe committed:
Porting to ddr

Still a problem concerning the node handle name of the ddr 'server'

Patrick-Lascombe committed:
Changing filter node handle name

Patrick-Lascombe committed:
Finishing porting to ddr

Patrick-Lascombe committed:
Recreating normal node scan_to*

Patrick-Lascombe committed:
Renaming the nodelet to a more explicit name

Patrick-Lascombe committed:
Merge remote-tracking branch 'upstream/indigo-devel' into indigo-devel

Patrick-Lascombe commented:
The travis build may have failed because I added the nodelet dependencies such as ddr and nodelet. I sthere a way to solve this problem ?

jonbinney commented:
@Patrick-Lascombe this package is pretty low in the ros stack and i'd rather not have non-core dependencies like ddynamic reconfigure (regular dynamic reconfigure would probably be fine). Also the dynamic reconfigure changes shouldn't be in the same PR as the nodelet changes.

What were the performance problems you were seeing that you're solving with this PR? What CPU usage levels for what number of laser scans per second? And how much did that change after these changes to the code?

Patrick-Lascombe committed:
adding cloud_to_cloud filter

Patrick-Lascombe committed:
Change input ropic

Patrick-Lascombe committed:
Remapping topic

Patrick-Lascombe committed:
Creating shadow_mean_filter

Patrick-Lascombe committed:
Scan mean shadow working

Need to do more test and tuning

Patrick-Lascombe committed:
Delete debug msgs

Patrick-Lascombe committed:
Creating standard deviation filter

Patrick-Lascombe committed:
Merge branch 'indigo-devel' into median_shadow_filter

Patrick-Lascombe committed:
Merge branch 'median_shadow_filter' into indigo-devel

Patrick-Lascombe committed:
Implementing cloud to cloud nodelet

Need a little bit more work

Commit:
change cfg names to allow build install

Patrick-Lascombe committed:
Adding ddr to scan shadow filter

Patrick-Lascombe committed:
Delete remap and modify intensity threshold

Patrick-Lascombe committed:
Adding manager to nodelet_example.launch

Patrick-Lascombe committed:
Cleaning nodelet compilation into only one library

Patrick-Lascombe committed:
Hot fix for angular bound filter

The filter does not clean some point behind the robot for some reason. When going from 4800/4 to 4800 the points not cleared increased

Patrick-Lascombe committed:
Comparing angles to PI instead of 3.14

Commit:
bump cmake

Patrick-Lascombe committed:
fix nodelet

Patrick-Lascombe commented:
Sorry for letting this PR aside for this long.
Since the default branch has been moved to kinetic-devel, I've sync my fork with the upstream and moved my modification to the kinetic-devel branch.
Hence I'll close this PR and open another one on the correct branch, with the modifications asked and the CPU usage levels
